File: esercizi_java/NetBeansProjects/Universita/src/universita/Studente.java
package universita;
import java.util.List;
import java.util.LinkedList;

/**
 *
 * @author tobiasacchetto
 */
public class Studente {
    protected int cod;
    protected String nome;
    protected List<Libretto> corso;
    public Studente(int cod,String nome){
        this.cod=cod;
        this.nome=nome;
        this.corso= new LinkedList<Libretto>();
    }
    
    public void addCorso(Corso c,String voto){
        corso.add(new Libretto(c,voto));
    }
    
    public double media(){
        double a=0;
        int z=0;
        for(Libretto l: corso){
            if(l.voto.equals("30L"))
                a=a+31;
            else
                a=Integer.parseInt(l.voto)+a;
            z++;
        }
        a=a/z;
        return a;
    }
    
    public String max(){
        double a=0;
        String nome="";
        for(Libretto l: corso){
            if(l.voto.equals("30L")){
                return l.c.nomecorso+"\t"+l.voto;
            }else
            {
                if(Double.parseDouble(l.voto)>a){
                    a=Double.parseDouble(l.voto);
                    nome=l.c.nomecorso;
                }
            }
        }
        return nome+"\t"+a;
    }
    
    public String toString(){
        return nome+"\t"+media()+"["+corso.toString()+"]";
    }
}

File: esercizi_python/compito_python_4/Studente.py
from Libretto import Libretto


class Studente:
    def __init__(self, cod, nome):
        self._cod = cod
        self._nome = nome
        self._corso = []

    def addCorso(self, l):
        self._corso.append(l)

    def media(self):
        a = 0.00
        z = 0
        for l in self._corso:
            if l._voto == "30L":
                a = a+31
            else:
                a = a+float(l._voto)
            z = z+1
        a = a/z
        return a
    

    def max(self):
        a=0.00
        nome=""
        for l in self._corso:
            if l._voto == "30L":
                return l._voto + "\t" + l._cod._nome
            else:
                if a<float(l._voto):
                    a=float(l._voto)
                    c = l
        return c
    





    def libret(self):
        res = ""
        for l in self._corso:
            res = res + l.toString()
        return res

    def toString(self):
        ciao = self.media()
        return self._nome + "\t" + str(self.media())+"["+self.libret()+"]"



